ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py

```diff
@@ -1611,3 +1611,71 @@ def test_tucker_tensorly(tol_norm_2, tol_max_abs, shape, ranks):
         ivy.max(ivy.abs(rec_svd - rec_random)) < tol_max_abs,
         "abs norm of difference between svd and random init too high",
     )
+
+
+# dot
+@st.composite
+def _generate_dot_dtype_and_arrays(draw):
+    shape_a = draw(
+        helpers.get_shape(
+            min_dim_size=2, max_dim_size=5, min_num_dims=0, max_num_dims=5
+        )
+    )
+    shape_b = draw(
+        helpers.get_shape(
+            min_dim_size=2, max_dim_size=5, min_num_dims=0, max_num_dims=5
+        )
+    )
+    # pdb.set_trace()
+    shape_a = list(shape_a)
+    shape_b = list(shape_b)
+    if len(shape_a) == 1 and len(shape_b) == 1:
+        shape_b[0] = shape_a[0]
+    elif len(shape_a) == 2 and len(shape_b) == 2:
+        shape_b[0] = shape_a[1]
+    elif len(shape_a) >= 2 and len(shape_b) == 1:
+        shape_b[0] = shape_a[-1]
+    elif len(shape_a) >= 1 and len(shape_b) >= 2:
+        shape_a[-1] = shape_b[-2]
+
+    dtype_1, a = draw(
+        helpers.dtype_and_values(
+            shape=shape_a,
+            available_dtypes=helpers.get_dtypes("float"),
+            min_value=-10,
+            max_value=10,
+        ).filter(lambda x: "float16" not in x[0])
```

Comment:
I don't follow why we need to add this condition? Can you please explain this? If `float16` is not supported by any of the backends, then we should be adding it to the `unsupported_dtypes` decorator in the backend implementation.

Reply:
added the supported_dtype to paddle and removed the condition